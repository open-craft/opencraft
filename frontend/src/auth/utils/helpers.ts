import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
