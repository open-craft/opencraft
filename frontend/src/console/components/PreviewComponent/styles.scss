.theme-preview {
  border: 1px solid black;
  background: white;

  .theme-preview-navigation {
    .row {
      margin: 0;
      padding-left: 10px;

      .col-md-3 {
        padding-right: 0;
        padding-left: 0;
      }
    }

    .customizable-link {
      height: 30px;
      font-size: 10px;
    }

    .customizable-button {
      width: 60px;
      height: 18px;
      font-size: 9px;
    }
  }

  .theme-home {
    .theme-hero-container {
      margin-top: 10px;
      padding-left: 10px;
      padding-right: 10px;

      .outer-wrapper {
        padding-top: 20px;
        padding-bottom: 20px;
      }

      .title {
        height: 50px;
        width: 50%;
        padding: 10px;
        margin: 0 auto;

        h1 {
          font-size: 0.7em;
        }

        p {
          font-size: 0.4em;
        }
      }
    }

    .theme-courses-container {
      padding-left: 30px;

      .theme-courses-item {
        padding: 10px 10px 10px 10px;
      }
    }
  }

  .theme-footer {
    .custom-footer {
      margin-bottom: 0;

      .footer-top-links {
        padding-left: 2px;
      }

      .col-md-4 {
        padding-left: 0;
      }

      .customizable-link {
        font-size: 8px;
      }

      .copyright {
        font-size: 6px;
      }

      .logo {
        img {
          max-width: 30px;
        }
      }

      .openedx-logo {
        img {
          max-width: 50px;
        }
      }

      .row.legal {
        .col:first-of-type {
          padding-left: 4px;
        }

        .customizable-link {
          font-size: 6px;
        }
      }
    }
  }
}
