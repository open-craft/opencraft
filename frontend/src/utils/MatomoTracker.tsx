import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
