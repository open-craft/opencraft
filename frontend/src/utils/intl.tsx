import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
