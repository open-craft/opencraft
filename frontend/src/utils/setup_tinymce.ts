import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
