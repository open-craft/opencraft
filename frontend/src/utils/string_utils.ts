export const toCamelCase = (s: string) => {
  return s.replace(/([-_][a-z])/gi, ($1) => {
    return $1.toUpperCase().replace('-', '').replace('_', '');
  });
};

export const sanitizeErrorFeedback = (input: {
  [key: string]: Array<string>;
}) => {
  // Loop at each error message and join them.
  // Also convert keys from snake_case to camelCase
  const newObject: any = {};

  Object.keys(input).forEach((key) => {
    newObject[toCamelCase(key)] = input[key].join();
  });

  return newObject;
};

export const capitalizeFirstLetter = (str: string) => {
  return str[0].toUpperCase() + str.slice(1);
};
