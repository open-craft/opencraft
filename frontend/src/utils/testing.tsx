import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
