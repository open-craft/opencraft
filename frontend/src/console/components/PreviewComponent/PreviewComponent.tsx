import * as React from 'react';
import './styles.scss';
import { Col, Row } from 'react-bootstrap';
import { NavigationMenu } from '../NavigationMenu';
import { InstanceSettingsModel } from '../../models';
import { HeroPreview } from '../../../ui/components/HeroPreview';
import { FooterPreview } from '../FooterPreview';
import { CoursesListingItem } from '../CoursesListingItem';

interface PreviewComponentProps {
  children?: React.ReactNode;
  instanceData: InstanceSettingsModel;
}

export const PreviewComponent: React.FC<PreviewComponentProps> = (
  props: PreviewComponentProps
) => {
  const { instanceData } = props;
  const themeData = instanceData.draftThemeConfig!;

  return (
    <div className="theme-preview">
      <Row className="theme-preview-navigation">
        <Col className="theme-preview-navigation">
          <NavigationMenu instanceData={instanceData} themeData={themeData} />
        </Col>
      </Row>
      <div className="theme-home">
        <Row className="theme-hero-container">
          <Col>
            <HeroPreview
              heroCoverImage={instanceData.heroCoverImage || ''}
              homePageHeroTitleColor={themeData.homePageHeroTitleColor}
              homePageHeroSubtitleColor={themeData.homePageHeroSubtitleColor}
              homepageOverlayHtml={
                instanceData!.draftStaticContentOverrides &&
                instanceData.draftStaticContentOverrides.homepageOverlayHtml
              }
            />
          </Col>
        </Row>
        <Row className="theme-courses-container">
          <Col md={3} className="theme-courses-item">
            <CoursesListingItem themeData={themeData} />
          </Col>
        </Row>
      </div>
      <Row className="theme-footer">
        <Col>
          <FooterPreview instanceData={instanceData} themeData={themeData} />
        </Col>
      </Row>
    </div>
  );
};
