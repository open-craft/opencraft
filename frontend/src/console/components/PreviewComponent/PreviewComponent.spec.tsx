import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { PreviewComponent } from './PreviewComponent';

it('renders without crashing', () => {
    const instanceData = {
        id: 1,
        instanceName: "test",
        subdomain: "test",
        publicContactEmail: "",
        privacyPolicyUrl: "",
        draftThemeConfig: {},
        draftStaticContentOverrides: {
          homepageOverlayHtml: "",
        },
        heroCoverImage: "test"
    }
    const tree = setupComponentForTesting(<PreviewComponent instanceData={instanceData}/>).toJSON();
    expect(tree).toMatchSnapshot();
});
