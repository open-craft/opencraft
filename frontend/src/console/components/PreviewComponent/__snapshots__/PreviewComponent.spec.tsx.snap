// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="theme-preview"
>
  <div
    className="theme-preview-navigation row"
  >
    <div
      className="theme-preview-navigation col"
    >
      <div
        className="main-navigation-menu navigation-menu row"
        style={
          Object {
            "background": undefined,
          }
        }
      >
        <div
          className="logo col-md-1"
        />
        <div
          className="col-md-1"
        >
          <button
            className="customizable-link"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "borderBottomColor": undefined,
                "color": undefined,
              }
            }
            type="button"
          >
            Courses
          </button>
        </div>
        <div
          className="col-md-3"
        >
          <button
            className="customizable-link"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "borderBottomColor": undefined,
                "color": undefined,
              }
            }
            type="button"
          >
            Discover New
          </button>
        </div>
        <div
          className="col-md-7"
        >
          <div
            className="navigation-menu navigation-submenu row"
          >
            <div
              className="col"
            >
              <button
                className="customizable-button"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "background": undefined,
                    "borderColor": undefined,
                    "color": undefined,
                  }
                }
                type="button"
              >
                Register
              </button>
              <button
                className="customizable-button"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "background": undefined,
                    "borderColor": undefined,
                    "color": undefined,
                  }
                }
                type="button"
              >
                Sign in
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="theme-home"
  >
    <div
      className="theme-hero-container row"
    >
      <div
        className="col"
      >
        <div
          className="hero-preview"
        >
          <div
            className="outer-wrapper"
            style={
              Object {
                "backgroundImage": "url(\\"test\\")",
              }
            }
          >
            <div
              className="title"
            >
              <div
                className="heading-group"
              >
                <h1
                  style={Object {}}
                >
                  
                </h1>
                <p
                  style={Object {}}
                >
                  
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="theme-courses-container row"
    >
      <div
        className="theme-courses-item col-md-3"
      >
        <div
          className="courses-listing-item row"
          style={
            Object {
              "borderBottomColor": undefined,
            }
          }
        >
          <div
            className="course-image col"
          >
            <img
              alt="course"
              src="course_image.jpg"
            />
          </div>
          <div
            className="organization col"
            style={
              Object {
                "color": undefined,
              }
            }
          >
            edX
          </div>
          <div
            className="course-number col"
            style={
              Object {
                "color": undefined,
              }
            }
          >
            DemoX
          </div>
          <div
            className="course-name col"
          >
            <button
              className="customizable-link noHover"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "borderBottomColor": undefined,
                  "color": undefined,
                }
              }
              type="button"
            >
              <span>
                Demonstration Course
              </span>
            </button>
          </div>
          <div
            className="course-start col"
            style={
              Object {
                "color": undefined,
              }
            }
          >
            Starts: 
            Feb 5, 2013
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="theme-footer row"
  >
    <div
      className="col"
    >
      <div
        className="custom-footer"
        style={
          Object {
            "background": undefined,
          }
        }
      >
        <div
          className="row"
        >
          <div
            className="col-md-4"
          >
            <div
              className="footer-top-links row"
            >
              <div
                className="col"
              >
                <button
                  className="customizable-link noHover"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "borderBottomColor": undefined,
                      "color": undefined,
                    }
                  }
                  type="button"
                >
                  <span>
                    About
                  </span>
                </button>
              </div>
              <div
                className="col"
              >
                <button
                  className="customizable-link noHover"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "borderBottomColor": undefined,
                      "color": undefined,
                    }
                  }
                  type="button"
                >
                  <span>
                    Blog
                  </span>
                </button>
              </div>
              <div
                className="col"
              >
                <button
                  className="customizable-link noHover"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "borderBottomColor": undefined,
                      "color": undefined,
                    }
                  }
                  type="button"
                >
                  <span>
                    Contact
                  </span>
                </button>
              </div>
              <div
                className="col"
              >
                <button
                  className="customizable-link noHover"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "borderBottomColor": undefined,
                      "color": undefined,
                    }
                  }
                  type="button"
                >
                  <span>
                    Donate
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            className="col-md-5"
          />
          <div
            className="openedx-logo col-md-3"
          >
            <img
              alt="Open edX logo"
              src="openedx-logo-footer.png"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="logo col-md-1"
          />
        </div>
        <div
          className="row"
        >
          <div
            className="copyright col"
            style={
              Object {
                "color": undefined,
              }
            }
          >
            Â© test. 
            All rights reserved except where noted. edX, Open edX and their respective logos are registered trademarks of edX Inc.
          </div>
        </div>
        <div
          className="legal row"
        >
          <div
            className="col"
          >
            <button
              className="customizable-link noHover"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "borderBottomColor": undefined,
                  "color": undefined,
                }
              }
              type="button"
            >
              <span>
                Privacy Policy
              </span>
            </button>
          </div>
          <div
            className="separator col"
          >
            -
          </div>
          <div
            className="col"
          >
            <button
              className="customizable-link noHover"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "borderBottomColor": undefined,
                  "color": undefined,
                }
              }
              type="button"
            >
              <span>
                Terms of Service
              </span>
            </button>
          </div>
          <div
            className="separator col"
          >
            -
          </div>
          <div
            className="col"
          >
            <button
              className="customizable-link noHover"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "borderBottomColor": undefined,
                  "color": undefined,
                }
              }
              type="button"
            >
              <span>
                Honor Code
              </span>
            </button>
          </div>
          <div
            className="separator col"
          >
            -
          </div>
          <div
            className="col"
          >
            <button
              className="customizable-link noHover"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "borderBottomColor": undefined,
                  "color": undefined,
                }
              }
              type="button"
            >
              <span>
                Take free online courses at edX.org
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
